plugins/org.python.pydev.ast/src/org/python/pydev/ast/codecompletion/revisited/visitors/FindDefinitionModelVisitor.java

```diff
@@ -391,6 +380,28 @@ public Object visitAssign(Assign node, int unpackPos) throws Exception {
         return super.visitAssign(node);
     }
 
+    /**
+     * @see org.python.pydev.parser.jython.ast.VisitorBase#visitNamedExpr(org.python.pydev.parser.jython.ast.NamedExpr)
+     */
+    @Override
+    public Object visitNamedExpr(NamedExpr node) throws Exception {
+        ILocalScope scope = new LocalScope(nature, this.defsStack);
+        scope.setFoundAtASTNode(node);
+        if (foundAsDefinition && !scope.equals(definitionFound.scope)) { //if it is found as a definition it is an 'exact' match, so, we do not keep checking it
+            return null;
+        }
+        if (tokenToFind.equals(NodeUtils.getFullRepresentationString(node.target)) && node.value != null) {
+            String rep = NodeUtils.getFullRepresentationString(node.value);
+            if (rep != null && !rep.isEmpty()) {
+                definitions
+                        .add(getAssignDefinition(new Assign(null, node.value, null),
```

Comment:
I think the only thing missing here is that you actually have the target in the `NamedExpr`, so, when constructing the `Assign` you should also pass the target.

i.e.:
`new Assign(new exprType[] {node.target}, node.value, null)`